web_ui/src/components/UsageBillingPage.tsx

```diff
@@ -749,6 +753,210 @@ function SubscriptionUpsellPrompt({
+    readonly country?: string
+  }
+  readonly className?: string
+}) {
+  const [line1, setLine1] = React.useState(defaultValue.line1 ?? "")
+  const [line2, setLine2] = React.useState(defaultValue.line2 ?? "")
+  const [city, setCity] = React.useState(defaultValue.city ?? "")
+  const [state, setState] = React.useState(defaultValue.state ?? "")
+  const [postalCode, setPostalCode] = React.useState(
+    defaultValue.postalCode ?? "",
+  )
+  const [country, setCountry] = React.useState(defaultValue.country ?? "")
+
+  const [apiState, updateStripeCustomerInfo] = useTeamApiMutation(
+    Current.api.updateStripeCustomerInfo,
+  )
+  function handleSubmit(e: React.ChangeEvent<HTMLFormElement>) {
+    e.preventDefault()
+    updateStripeCustomerInfo({
+      address: { line1, line2, city, state, postalCode, country },
+    })
+  }
+  return (
+    <Card className={className}>
+      <Card.Body>
+        <Card.Title>Postal Address</Card.Title>
+        <Form onSubmit={handleSubmit}>
+          <Form.Group>
+            <Form.Control
+              type="text"
+              placeholder="Address line 1"
+              value={line1}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                setLine1(e.target.value)
+              }
+            />
+          </Form.Group>
+          <Form.Group>
+            <Form.Control
+              type="text"
+              placeholder="Address line 2"
+              value={line2}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                setLine2(e.target.value)
+              }
+            />
+          </Form.Group>
+          <Form.Group>
+            <Form.Control
+              type="text"
+              placeholder="City"
+              value={city}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                setCity(e.target.value)
+              }
+            />
+          </Form.Group>
+          <Form.Group>
+            <Form.Row>
+              <Col>
+                <Form.Control
+                  type="text"
+                  placeholder="State / Province / Region"
+                  value={state}
+                  onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                    setState(e.target.value)
+                  }
+                />
+              </Col>
+              <Col>
+                <Form.Control
+                  type="text"
+                  placeholder="ZIP / Postal Code"
+                  value={postalCode}
+                  onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                    setPostalCode(e.target.value)
+                  }
+                />
+              </Col>
+            </Form.Row>
+          </Form.Group>
+          <Form.Group>
+            <Form.Control
+              type="text"
+              placeholder="Country"
+              value={country}
+              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
```

Comment:
We have to type all of these params for some reason. Using a regular `<input/>` element doesn't require that.